Data/Map/Lazy.hs

```diff
@@ -14,19 +14,63 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- An efficient implementation of ordered maps from keys to values
--- (dictionaries).
 --
--- API of this module is strict in the keys, but lazy in the values.
--- If you need value-strict maps, use "Data.Map.Strict" instead.
--- The 'Map' type itself is shared between the lazy and strict modules,
--- meaning that the same 'Map' value can be passed to functions in
--- both modules (although that is rarely needed).
+-- = Lazy Maps (aka dictionaries)
```

Comment:
Do not put the "aka dictionaries" in the title. Also don't write it that way anywhere else. Also, we don't actually *have* strict and lazy `Map`s. We have a single `Map` type offering lazy and strict *operations*. The title should probably be "Finite Maps (lazy interface)" or something like that.